src/repr/src/row.rs

```diff
@@ -294,51 +307,47 @@ unsafe fn read_lengthed_datum<'a>(data: &'a [u8], offset: &mut usize, tag: Tag)
 /// This function is safe if a `Datum` was previously written at this offset by `push_datum`.
 /// Otherwise it could return invalid values, which is Undefined Behavior.
 unsafe fn read_datum<'a>(data: &'a [u8], offset: &mut usize) -> Datum<'a> {
-    let tag = read_copy::<Tag>(data, offset);
+    let tag = Tag::try_from_primitive(read_byte(data, offset)).expect("unknown row tag");
```

Comment:
Let’s check that this try/expect gets rolled into the match rather than producing an extra branch.